### PR TITLE
Fixed bug where HotKeys are not being read from the config file

### PR DESCRIPTION
**Pull request by LGM-Doyle**
- Removed the shadowing of key_modkey_pair in the try catch loop
- Added debug message reporting which hotkeys have been added
- Fixed HotKeyFromString to allow more than two modifier keys
- Changed behavior to make whitespace not matter in the config string

### CODE REVIEW

File: UI/Hotkeys.cpp [hunk 1]
```diff
@@ -107,28 +108,33 @@ std::pair<GG::Key, GG::Flags<GG::ModKey> > Hotkey::HotkeyFromString(const std::s
...
+    // Strip whitespace
+    std::string copy = str;
+    copy = std::string(copy.begin(), std::remove_if(copy.begin(), copy.end(), isspace));
+
+    size_t plus = copy.find('+');
+    bool hasModifier = plus != std::string::npos;
+
     GG::Flags<GG::ModKey> mod = GG::MOD_KEY_NONE;
-    if (plus != std::string::npos) {
+    if (hasModifier) {
```
Comment (geoffthemedio) [inline UI/Hotkeys.cpp]: please use lowercase_underscore_separated_variable_names

File: UI/Hotkeys.cpp [hunk 2]
```diff
@@ -233,6 +239,9 @@ void Hotkey::ReadFromOptions(OptionsDB& db) {
 
         hotkey.m_key = key_modkey_pair.first;
         hotkey.m_mod_keys = key_modkey_pair.second;
+
+        DebugLogger()  <<__func__ << "Added hotkey '" << hotkey.m_key  << "' with modifiers '"
```
Comment (geoffthemedio) [inline UI/Hotkeys.cpp]: I think it was concluded that **func** was not portable, so if you really need it, write it out explicitly.

File: UI/Hotkeys.cpp [hunk 3]
```diff
@@ -233,6 +239,9 @@ void Hotkey::ReadFromOptions(OptionsDB& db) {
...
+                       <<hotkey.m_mod_keys << "' for hotkey '" << hotkey.m_name<<"'";
```
Comment (geoffthemedio) [inline UI/Hotkeys.cpp]: single spaces around the << please

**Comment by geoffthemedio** [comment]
Seems to work for me. If you have a name and alternate alias for the credits, please provide them.

**Comment by Vezzra** [comment]
@LGM-Doyle, we also need a license statement from you, that you release your contribution under the GPL 2.0 and later license (according to the licenses used by the FreeOrion project).

It would be best if you would give a general statement that not only refers to this contribution here, but is formulated in a way that it encompasses all your contributions to the project. You can find the license statements of the FreeOrion project [here](http://freeorion.org/index.php/FreeOrionWiki:Copyrights), [here](https://github.com/freeorion/freeorion/blob/master/default/COPYING) and [here](https://github.com/freeorion/freeorion/blob/master/README.md#license).

Please post a respective statement on [our forums](http://www.freeorion.org/forum/) (e.g. the [General Discussion](http://www.freeorion.org/forum/viewforum.php?f=2&sid=3e52a3fce10ac7c356e4ac3559c6359a) section).

Unless you've done so already and I'm not aware of it (in this case please point me to your statement). Or maybe you're already on our forum under a different username? In this case add a short comment to your forum sig stating that your username on github is LGM-Doyle, so that the connection is clear.

**Comment by geoffthemedio** [comment]
I figured it was OK to assume that FreeOrion repository code pull requests on github were GPL 2.0 licensed. I insisted on such a statement on the forums as it was not directly clear there under what license the code provided was expected to be released...

**Comment by Vezzra** [comment]
Oh, ok, didn't know that... are you sure? I mean, I thought that we'd at least had to have some statements like "If you make pull requests for our repo you agree to release your contribution under the licenses in use by the project" in the README.md or a similar prominent location.

Can we safely assume implicit agreement to our licenses without such a statement anywhere? Not that I really have any idea what I'm talking about, all that law stuff is beyond me...

**Comment by geoffthemedio** [comment]
Not sure, no. But in this case, the request also modifies existing code, which was GPL licensed, and republishes it in a new repository on GitHub,

https://github.com/LGM-Doyle/freeorion/

The only way he could do this is by releasing it under compatible license(s). Also, that page has the same license statement from the Readme.md file, which indicates the code is GPL 2.0.

**Comment by Vezzra** [comment]
Ah yes, you're right of course, didn't think of that. Working with a clone of our repo means copying the license statements, so I guess we can safely assume such contributions to be released under the same licenses.

**Pull request merged**